pyspedas/utilities/download.py

```diff
@@ -399,12 +530,30 @@ def download(
 
         short_path = local_file[: 1 + local_file.rfind("/")]
 
+        if is_fsspec_uri(url) and no_download:
```

Comment:
If I'm understanding correctly, this would catch when the remote is an s3 path (like Heliocloud holdings) and the user DOES NOT want to download, but the code underneath if the user wants to download locally from s3 holdings. I think the if statement should be `if is_fsspec_uri(url) and not no_download:`. 

This case isn't well tested since we don't have any remote s3 holdings hooked up yet, but I think it's good we have this set up when we do move on to that step.

Reply:
I'm wondering if we should default to not downloading from the URI for either `no_download` case, and then adding another keyword flag (or worse environment variable 😨) when the user actually download from s3, since it should be a very rare case that we do not want to encourage. The reason I say this is because sometimes `no_download` is called `no_update` in some calls and it's hard to interpret what would be acceptable to the user in this case.
- `if is_fsspec_uri(url) and not no_download` - remote is URI and the user wants to download locally. Should default to stream from s3 unless user REALLY wants to download from s3 for some reason.
- `if is_fsspec_uri(url) and no_download` - remote is URI and the user does not want to download. Stream data from s3.
- `if is_fsspec_url(url) and not no_download and [SOME FLAG]` - remote is URI and user really REALLY wants to download locally despite our pleas for them not to.